### ISSUE TITLE
generate json file from val set in M4C and only get 33.82 in evalutation

### ISSUE THREAD

**Issue by akira-l** [facebookresearch/mmf#534]
## ❓ Questions and Help
I generated the json file from val set via evaluation_loop.py. (copied the code in test condition to the val condition) 
Details: 
I copied these code into val condition and got the json file. 
`for batch in tqdm.tqdm(dataloader):
                    prepared_batch = reporter.prepare_batch(batch)
                    prepared_batch = to_device(prepared_batch, torch.device("cuda"))
                    model_output = self.model(prepared_batch)
                    report = Report(prepared_batch, model_output)
                    reporter.add_to_report(report, self.model)`
 

I analysed the results, but only achieved 33.82 in VQA evaluation. Why ?

**Comment by akira-l** [comment]
I did not download the st-vqa feature for training, and only used the textvqa features in config file. So I use the path in my default path of textvqa. 

the yaml config in joint_with_stvqa.yaml is 
`dataset_config:
  textvqa:
    use_images: false
    use_features: true
    zoo_requirements:
    - textvqa.defaults
    - textvqa.ocr_en
    - stvqa.defaults
    - stvqa.ocr_en
    features:
      train:
      - textvqa/defaults/features/open_images/detectron.lmdb,textvqa/ocr_en/features/ocr_en_frcn_features.lmdb
      - textvqa/defaults/features/open_images/detectron.lmdb,textvqa/ocr_en/features/ocr_en_frcn_features.lmdb
    annotations:
      train:
      - textvqa/defaults/annotations/imdb_train_ocr_en.npy
      - stvqa/defaults/annotations/imdb_subtrain.npy
`

**Comment by ronghanghu** [comment]
Hi @akira-l, to debug the issue, could you first verify that the local evaluation results matches the expected accuracy? In particular, if you run the local evaluation script below using the released model, do you get the expected 0.4065 accuracy?

Evaluation script on the TextVQA validation set:
```
mmf_run config=projects/m4c/configs/textvqa/defaults.yaml \
    datasets=textvqa \
    model=m4c \
    run_type=val \
    checkpoint.resume_zoo=m4c.textvqa.with_stvqa
```

Expected output:
```
val/total_loss: 6.3802, val/textvqa/m4c_decoding_bce_with_mask: 6.3802, val/textvqa/textvqa_accuracy: 0.4065
```

**Comment by akira-l** [comment]
I get the similar problem in [https://github.com/facebookresearch/mmf/issues/535](url) when operating the original evaluation scripts.

**Comment by akira-l** [comment]
same to the #535
